lib/ruby_indexer/lib/ruby_indexer/declaration_listener.rb

```diff
@@ -660,10 +660,21 @@ def collect_comments(node)
 
       comments = +""
 
-      start_line = node.location.start_line - 1
-      start_line -= 1 unless @comments_by_line.key?(start_line)
+      comment_lines = [node.location.start_line - 1, node.location.start_line - 2]
 
+      comment_lines.each do |line_number|
+        collect_comment_lines(line_number, comments)
+      end
+
+      comments.chomp!
+      comments
+    end
```

Comment:
@vinistock, Even though I adjusted the code (I made a slight change because @source_lines is an array, which means the index starts with 0 instead of 1), the `test_completion_resolve_with_owner_present` test still failed. It makes me wonder whether this test was designed wrongly.  

```ruby
start_line = node.location.start_line - 1
start_line -= 1 unless ... || (@source_lines[start_line - 1] && !@source_lines[start_line - 1].strip.empty?)
```

Reply:
I just took a better look at the test and you are completely right, the test is wrong and we didn't catch because of the bug you are fixing.

The `Foo!` comment should be on top of the instance variable and not on top of the `initialize` method.

Reply:
Yeah, I still think that the approach with `@source_lines` is a bit simpler than the current one. If you don't mind updating to that, then we can ship this one!